### ISSUE TITLE
Bringing back the LAYERS_EASILY flag.

### ISSUE THREAD

**Issue by Rivet-the-Zombie** [CleverRaven/Cataclysm-DDA#12741]
Wearing a bra and tank top and trenchcoat causes encumbrance stacking penalties (seventeen encumbrance on the torso when combined) now, which is excessive. Cargo pants and boxer shorts come out to seven, which also seems a bit high when you consider that adding any further equipment worn on the legs will bring immediate penalties.

Maybe we could add an underwear layer and/or reduce layering penalties a bit in order to fix this. Does anyone else have ideas for how to address this?

**Comment by Coolthulhu** [comment]
Since fitting items no longer reduces encumbrance to 0, penalties for layering could be scaled off encumbrance.
Something like having every item on the layer other than the most encumbering cause penalty equal to its encumbrance value.

For example, when wearing 4 items (on the same slot/layer) with encumbrances 5, 5, 3 and 2, the layering penalty would be 10 (5+3+2), for total encumbrance of 25.

**Comment by DeadLeaves** [comment]
While both point out great ways to fix it, I think an underwear layer could be good so there is no outright negative to playing a female character. (Bra vs no bra)

**Comment by codemastergh** [comment]
The layering penalty scaling with the encumbrance is a good idea.

**Comment by kevingranade** [comment]
Something isn't adding up with the examples, bra and tank top are in the same layer, but trenchcoat is not.
Likewise boxer shorts and cargo shorts shouldn't be interacting at all.

**Comment by kevingranade** [comment]
I'm not a fan of another "underwear" layer, that's what the "under" layer is for already.
I'd totally buy that a large number of encumbrance numbers need to get bumped around, and that +10 enc for each item after the first in a layer is excessive.
So for Coolthulu's suggestion, instead of +10 per article after the first, it's +enc of each item after the first (with 'first' being which ever has the highest enc, OR it could be the outermost article to encourage sensible layering).  That sounds good to me along with enforcing a minimum of 1 enc everywhere, otherwise there's a perverse incentive to find and stack different clothing with no enc.

**Comment by kevingranade** [comment]
To widen the discussion a touch, the only guideline we have is "normal street clothes shouldn't cause encumbrance".  I always took that to mean non-winter street clothes, no fair trying to get everyday wear in Alaska dropped to < 10 :)
Even just a set of normal ensembles would work, and I could whip up a unit test to verify that the various ensembles stay where they're expected.  The ensembles would be simple lists of articles, lets assume fitted.

**Comment by wmafie** [comment]
I like the idea of sensible layering. As of now, I just rearrange my clothes according to how rare/important they are, in descending order.

**Comment by DavidKeaton** [comment]
I layer it based on what should be where, because I'm compulsive like that. (anyone who dares look at my code can perceive this ;-)

**Comment by DeNarr** [comment]
I second the idea of basing the penalty off of the encumbrance of the items equipped. I find myself looking at stuff like the Light Survivor Harness (enc 1 fits), and thinking "Well, you're going to add 11 enc anyways, so might as well just grab something bigger"

**Comment by Cat09** [comment]
could layering be affected by the material thickness? that might give more reason for increased granularity in that field though.

**Comment by kevingranade** [comment]
It could, but why? We already have a value that indicates how bulky it is.

**Comment by Killzemallgod** [comment]
A few more layers could help, underwear and stockings conflict each other due to underwear being legs. seeing as underwear take up 15% coverage and stockings 75% wouldn't it be more ideal to take that into account before doing the plently for stacking too much on one layer?

Innermost vs outermost should be applied so to avoid people wanting to protect something that logicaly should be on the correct layer. such as putting a backpack furtherup to avoid it taking damage.

The way encumbrance works now makes it pretty hard to melee dogs and other small monsters if you got too much on the torso, torso makes more sense for unarmed melee attacks while the weapons would use just hands and arms depending on their volume.

**Comment by Man-animal** [comment]
I don't think it's worth adding more layers to keep track of (and more potential balance issues) solely to smooth a few edge cases that are more or less irrelevant once you've looted a few dressers.  Scaled stacking penalties seem like a far more elegant and universal solution.

**Comment by kevingranade** [comment]
I'm very much against adding more layers, especially if the argument is
simply that a few things are causing high penalties.
That is a sign that there's a problem, but I think Coolthulu's suggestion
is a much better approach, e.g. if you layer two enc 0 items, the effective
encumbrance will be 1, and +1 for each additional enc 0 item.
I think with the granularity the system has now, that ends up being
reasonable, if you wear 10 pairs of stockings it's going to start adding up
at some point.  Probably well before 10, but if that turns out to be a
problem we can take another look at it.

**Comment by Killzemallgod** [comment]
Coverage seems like a logical way to work out how much the stacking penalties are applied.
Using coverage as value to determine if over 100 in total on the bodypart on that layer then add ecumbrance, might even be able to use volume and weight if its needing to go further.

Though volume seems like it might also help in cases of socks and other thing.
Ecumbrance Plenty = X x Items that share a layer.
If Bp layer Coverage > 100 then (layer Coverage/10) x Volume = SP
then SP x Ecumbrabce Plenty 

Though you might want to have it so it also takes into account if an items spans more then one bp like stockings so it doesn't conflict with underwear.
Need something to also make sure that everything is layered correctly or you gain ecumbrance. Like wearing boxers on the outside of your clothing or bras over the body armor. Though does intermost vs outermost impact what is damaged first?

**Comment by kevingranade** [comment]
If we want resolution under 1 encumbrance, that would make sense, but no matter the coverage, is there really anything that you could wear 10 of and not have any effect?
As for your other formula, we already have a lot of complaints that encumbrance is hard to follow without making it factor in so many different values.
I don't understand why stocking shouldn't conflict with underwear, one goes on top of the other, yes?  It should be minimal, but with Coolthulu's suggestion you're talking 1 encumbrance for that combination, which IS minimal.
It came up before that any "fitted" bonus shouldn't apply if the clothing is outside its normal layer, that sounds like a good idea to me.
Yes, outermost items are hit first, inner layers are only hit if the attack penetrated the outside.

**Comment by Killzemallgod** [comment]
A penalty should be applied if 2 items covering the same area go past a total coverage of 100. then have a min of what would have ifs its past 100 as well for more then 2 items covering the same area. The idea wasn't really to get a resoultuion under 1 encombrance but to avoid getting one if theres 2 items that share the same area and their total coverage is under 100 shouldn't incur a penalty.
Could just get it so for 150 coverage multiplicates it (5+1) or what might be suitable so its not so hard to follow.
It does need to be made clear in the UI where the numbers are coming from 
encumbrance and Warmth 7+10 = 17 (-35). 

Is there 4 or 5 layers? Looking at the code theres seems to be 4, skintight, wasit, outer and belted though clothing appears to be its own outside of that.
Is warmth displayed in Fahrenheit or Celsius or just is own value based on imperial measurement?

**Comment by DavidKeaton** [comment]
warmth is more of an AUM than based on any one measurement standard. cold < 0 and hot > 29

**Comment by kevingranade** [comment]
On Jun 25, 2015 5:07 PM, "Imperator_Urist" <notifications@github.com> wrote:
>if theres 2 items that share the same area and their total coverage is
under 100 shouldn't incur a penalty.

I don't agree, coverage and encumbrance simply have nothing to do with each
other, and should not have anything to do with each other.
Coverage means one thing only, "If something hits this body part, how
likely is it to hit this clothing?"

**Comment by Frost-wood** [comment]
Perhaps it would be better if the encumbrance wasn't a flat ten but an exponent maybe at a base 2-you start off low, but then it increases exponentially as you layer.  It starts at a low 2^1, then 2^1+2^2, then 2^1+2^2+2^3 etc.

Or it could go 3+6+9+12+15+18...

**Comment by ArikVhedrovix** [comment]
+1 Vote for reducing the additional clothes penalty to less than the flat 10. That is really steep and makes it seems to me it makes it difficult for me to keep warm and be reasonably good at hitting things in melee.

**Comment by CapitalistShrub** [comment]
To me, the biggest issue with the current layering system is that it doesn't allow much flexibility - being limited to a single item on each layer before encumbrance has more-or-less forced me to optimize what I'm wearing and avoid certain articles all together; there is a lot more onus on maximizing volume while occupying the minimal number of layers.

The most egregious of these are quivers and holsters - the storage they effectively provide is quite low for the cost of the valuable strapped layer (excepting the ankle holster as I don't know if there are other strapped feet layer items). Unless there is some hidden benefit to holsters that I'm unaware of, a runner pack is 100% better than the back and shoulder holsters for identical encumbrance, while most belts (even a basic leather belt in some cases!) are better than any of the waist holsters or sheaths.

Less of an issue, but still noteworthy - I'm encountering more issues staying dry early game due to this system as well. Like ArikVhedrovix said, staying warm and light is difficult, and because of this it seems like waterproof items are just about the only way to go as getting the required warmth to resist water effectively doesn't happen. 

I'll throw some examples together with and without some of the proposed solutions here for comparison's sake.

**Comment by Coolthulhu** [comment]
Problems with rain? I'd say it's way too easy to just ignore it. Even early on, a trenchcoat kills rain (almost) completely.

**Comment by CapitalistShrub** [comment]
Well, short of low focus due to morale loss there isn't a significant cost for it at the moment, sure.

Ran up a real quick fringe case (to get the formatting right) looking at just torso encumbrance as it's the most likely to me to be affected or have balance issues with layering changes - all articles are fitted where possible.

Six test cases - Coolthulu's idea with and without dropping the highest, Coolthulu's idea with and without dropping with a min encumbrance of 1 for each item (layering penalty was always assumed to be at least 1), and Frost-wood's scalings of 2^n and 3n for articles after the first.

Out of time tonight, but if folks like the format or find it useful I can run more standard loadouts, with a handful more of the extremes sprinkled in for fun.

![thehumanbackpack](https://cloud.githubusercontent.com/assets/13109908/8612718/b057f172-269a-11e5-8c79-23f21cd021f4.png)

**Comment by karlnp** [comment]
Speaking only for myself, I'd like to see some more standard layouts to give me a reference point.

**Comment by Zireael07** [comment]
I like the table.

**Comment by CapitalistShrub** [comment]
I'm working on them karlnp, no worries! I think extremes are still worth looking at (high volume and high bash/cut protection) since they're what would affect balance the most, but I'm primarily putting together more typical loadouts.

**Comment by kevingranade** [comment]
Addressed by #13043
